src/Utils.re

```diff
@@ -16,3 +18,24 @@ let changeAtIndex = (index, replaceFunc, newValue, list) =>
      );
 
 let replaceFunc = (newValue, _) => newValue;
+
+let allRepsAreEqual = (sets: list(set)): bool =>
+  sets |> List.for_all(s => s.reps === List.hd(sets).reps);
+
+let formatSets = (sets: list(set)): string =>
+  if (allRepsAreEqual(sets)) {
+    (sets |> List.length |> string_of_int)
```

Comment:
I suggest we break it up into smaller functions - e.g. `formatSetsReps` and `formatCustomSets` as we do in the TS version. Also interesting whether it crashes with an empty array (`List.hd` does). I guess it depends on whether `allRepsAreEqual` returns true or false for an empty list.

Reply:
Then we can use `?` `:` syntax instead of if/else.